### PR TITLE
fix: add PR source validation to Jenkinsfile

### PR DESCRIPTION
**Pull request by mdeltito**
Semver: patch
Ref: LOG-7712

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,7 +1,8 @@
 library 'magic-butler-catalogue'
 
-def projectName = "eslint-config-logdna"
-def repo = "logdna/${projectName}"
+def PROJECT_NAME = "eslint-config-logdna"
+def REPO = "logdna/${PROJECT_NAME}"
+def TRIGGER_PATTERN = ".*@logdnabot.*"
```
Comment (esatterwhite) [inline Jenkinsfile:5]: do this need a global or multi-line switch?

Comment (mdeltito) [reply Jenkinsfile:5]: good point, it probably should

Comment (esatterwhite) [reply Jenkinsfile:5]: It looks like it passes the entire body of the comment. so this should be ok. It doesn't look like you can pass switches anyway.

**Review by esatterwhite — DISMISSED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (mdeltito) [reply Jenkinsfile:5]: actually yeah, doesn't matter because this isn't anchored

**Review by darinspivey — DISMISSED** [review]

**Review by evanlucas — DISMISSED** [review]

**Review by jakedipity — DISMISSED** [review]

File: Jenkinsfile [hunk 2]
```diff
@@ -99,7 +116,7 @@ pipeline {
         versioner(
           token: "${GITHUB_PACKAGES_TOKEN}"
         , dry: true
-        , repo: repo
+        , repo: REPO
```
Comment (ligerzero459) [inline Jenkinsfile:119]: Can you add `, branch: "master"` to the `versioner` blocks? The new version understands that and it'll set us up to change over to using `main` branch whenever we're ready

Comment (mdeltito) [reply Jenkinsfile:119]: that's still the default for now, right? we can probably just add it when we change the base branch name, unless there's a reason for doing it sooner

Comment (ligerzero459) [reply Jenkinsfile:119]: Just reducing the likelihood it'll be forgotten later. If you don't think it's a prudent course, I'll go with your judgement 😁

Comment (mdeltito) [reply Jenkinsfile:119]: ah yeah, mainly thinking that if we add it here, we should do it in all of these. I'll follow your lead 👍

File: Jenkinsfile [hunk 3]
```diff
@@ -11,7 +12,23 @@ pipeline {
...
+        expression { env.CHANGE_FORK }
+        not {
+          triggeredBy 'issueCommentCause'
+        }
+      }
+      steps {
+        error("A maintainer needs to approve this PR for CI by commenting")
+      }
+    }
+
```
Comment (esatterwhite) [inline Jenkinsfile:31]: Wonder if it is possible to move this into the magic butter libs so we don't have to keep coping this + the regex around.

A function that returns a stage.

Comment (mdeltito) [reply Jenkinsfile:31]: looking into this separately from these PRs, I think it'd be good to have more of these identical steps broken out into magical butler abstractions

Comment (jmoses) [reply Jenkinsfile:31]: I hereby submit a petition to rename the jenkins library `magic-butter` as a much more worthy name.

**Review by darinspivey — APPROVED** [review]

**Review by ligerzero459 — APPROVED** [review]

**Pull request merged**